src/bisect.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+/* @flow */
+
+var sign = require('./sign');
+/**
+ * [Bisection method](https://en.wikipedia.org/wiki/Bisection_method) is a root-finding 
+ * method that repeatedly bisects an interval to find the root.
+ * 
+ * This function returns a numerical approximation to the exact value.
+ * 
+ * @param {Function} f input function
+ * @param {Number} a start of interval
+ * @param {Number} b end of interval
+ * @param {Number} nMax the maximum number of iterations
+ * @param {Number} tol the error tolerance
+ * @returns {Number} estimated root value
+ * 
+ * @example
+ * bisect(Math.cos,0,4,100,0.003); // => 1.572265625
+ */
+function bisect(f/*: (x: any) => number */, a/*: number */, b/*: number */, nMax/*: number */, tol/*: number */)/*:number*/ {
```

Comment:
since `a` is the start and `b` is the end, can you name the variables `start` and `end`? Preferring clarity over terseness is the pattern in simple statistics. Also:

* tol -> errorTolerance
* nMax -> maxIterations